x-pack/plugins/observability/public/components/shared/core_web_vitals/core_vital_item.tsx

```diff
@@ -81,19 +82,22 @@ export function CoreVitalItem({
   ranks = [100, 0, 0],
   isCls,
   helpLabel,
+  hasVitals = false,
 }: Props) {
   const palette = euiPaletteForStatus(3);
 
   const [inFocusInd, setInFocusInd] = useState<number | null>(null);
 
   const biggestValIndex = ranks.indexOf(Math.max(...ranks));
 
-  if ((value === null || value !== undefined) && ranks[0] === 100 && !loading) {
+  if (!value && !hasVitals && !loading) {
```

Comment:
Both the check against `ranks[0] === 100` and `value === null || value !== undefined` where causing issues. For the latter, even when value was defined, it was hitting this condition because of the `value !== undefined`.

Reply:
As I understand it, this is to we have to check `value` and `hasVitals` because `value` could be `0`, for two reasons:

1. There's no data
2. There is data, but the actual value is zero

so, we check `hasVitals` to ensure that it's the second case before we display it. Is that right?

IMHO the bigger problem here is the API. This would all be simpler if it returned `null` for cases where CLS was invalid, and a number otherwise. The API [rounds the number and returns it as a string with a fixed number of decimal points](https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/server/lib/rum_client/get_web_core_vitals.ts#L135). 

I'd propose we go a bit deeper here and modify the API either return `null` if there is no CLS data, that way our UI can be simple.

Reply:
TO add a bit more detail, all you'll need to do on the server is remove the `toFixed` call.